File: Arrays/two sum.py
class Solution:
    def twoSum(self, nums, target):
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[i] + nums[j] == target:
                    return [i , j]
                


# optimal solution 

# Space: O(n)
# Time: O(n)


class Solution(object):
    def twoSum(self, arr, target):
        hmap ={}

        for i in range(len(arr)):
            addup = target - arr[i] 
            if addup in hmap:
                return [i,hmap[addup]]
            hmap[arr[i]] = i

File: Arrays/Java/Two Sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {

        Map<Integer,Integer> set = new HashMap<>();

        for(int i = 0; i < arr.length; i++){
            int addup = target - arr[i];
            if(set.containsKey(addup)){
                return new int[] {i , set.get(addup)};
            }
            set.put(arr[i] , i);

        }
        return new int[]{};
        
    }
}